Enable CUTTER_USE_BUNDLED_RIZIN by default (#2622)
rizinorg/cutter: docs/source/building.rst

 If you just want to use the latest Release version of Cutter, please note
 that we provide pre-compiled binaries for Windows, Linux, and macOS on
 our `release page <https://github.com/rizinorg/cutter/releases/latest>`_ and

>  `CI page <https://github.com/rizinorg/cutter/actions/workflows/ccpp.yml?query=branch%3Amaster>`_ (requires logging in) for latest development builds.

Reviewer: @karliss can we instead use https://nightly.link for linking dev builds? It doesn't require logging in and offers clean download page for all artifacts.
```suggestion
 `CI page <https://nightly.link/rizinorg/cutter/workflows/ccpp/master>`_ for latest development builds.
```

Reviewer 2: I am hesitant on giving links to small third party service as primary link for downloading executables. Users can use such services if they want but that would have to be their own choice.

* dev-builds aren't critical there is always option to use release package or compile yourself
* everyone participating in development or reporting issues already have a GitHub account
* If you don't have GitHub account because you don't trust them you shouldn't blindly trust nightly.link either especially without means of verifying origin of what you download.

> This page describes how to do basic build from command line. If you are planning to modify Cutter it is recommended to also read :doc:`development environment setup</contributing/code/ide-setup>`.

Author: ```suggestion
This page describes how to do a basic build from the command line. If you are planning to modify Cutter it is recommended to also read our :doc:`development environment setup</contributing/code/ide-setup>`.
```

   ./build/cutter

Making Linux distribution specific packages
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> When making a distribution specific packages the default install target should give you a good starting point.

Reviewer: ```suggestion
When making a distribution specific package, the default install target should give you a good starting point.
```

> It uses CMake builtin functionality and `GNUInstallDirs <https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html?highlight=gnu%20directories>`_ for

Author: ```suggestion
It uses CMake built-in functionality and `GNUInstallDirs <https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html?highlight=gnu%20directories>`_ for
```

> installing the executable, desktop file, headers and other files required for plugin compilation. See CMake documentation for adjusting installed file locations and properties.

Author: optional: link to cmake docs?

Reviewer 2: Are you referring to
> It uses CMake built-in functionality 

or 
> See CMake documentation

In both cases there is no single part of documentation to link since there at least a dozen different options controlling various aspects of file installation.

Author: the latter. Alright then :)

It shouldn't be necessary to manually copy files from plain build.

> It is recommended to build and package rizin as separate package so that it can be used with or without Cutter. Doing that will also give more control over the way rizin dependencies are handled. We are trying to maintain

Author: ```suggestion
It is recommended to build and package rizin as a separate package so that it can be used with or without Cutter. Doing that will also give more control over the way dependencies are handled. We are trying to maintain
```

compatibility with latest rizin release at the time of Cutter release and making a new Cutter release when new rizin version is released.

> If you are packaging Cutter users will appreciate if you also package `rz-ghidra <https://github.com/rizinorg/rz-ghidra>`_ and `jsdec <https://github.com/rizinorg/jsdec>`_ decompilers as optional packages.

Author: ```suggestion
If you are packaging Cutter, users will appreciate it if you also package `rz-ghidra <https://github.com/rizinorg/rz-ghidra>`_ and `jsdec <https://github.com/rizinorg/jsdec>`_ decompilers as optional packages.
```

It should be possible to compile Cutter plugins against proper Cutter installation without having direct access to Cutter source code.

> If name Cutter or cutter conflicts with other packages or their content, rz-cutter can be used.

Author: ```suggestion
If the names "Cutter" or "cutter" conflict with other packages or their content, "rz-cutter" can be used.
```

CMAKE options:

* ``-DCMAKE_BUILD_TYPE=Release`` turn on release optimizations, unless your distro has more specific guidelines for common compiler options.
* ``CUTTER_USE_BUNDLED_RIZIN=OFF`` turn off use of rizin from submodule to use previously packaged rizin. Note that ``CUTTER_USE_BUNDLED_RIZIN=ON`` doesn't install rizin in a way suitable for linux packaging without doing additional manual steps making packaging process more complex. Bundled rizin will also likely conflict with standalone rizin package.
* Correct install prefix. By default CMake will install to /usr/local suitable for user builds. Change it according to your distro packaging guidelines.
* ``CUTTER_ENABLE_PYTHON`` and  ``CUTTER_ENABLE_PYTHON_BINDINGS`` it is recommended to turn on for complete user experience. May require manual path specification on distros with multiple python versions.
* ``CUTTER_ENABLE_GRAPHVIZ`` and ``CUTTER_ENABLE_KSYNTAXHIGHLIGHTING`` optional but nice to have since they are available on most distros.

> * ``CUTTER_EXTRA_PLUGIN_DIRS`` use it to specify additional plugin search locations if distro packaging guidelines require you placing them in locations Cutter doesn't use by default.

Author: @karliss, duplication? (lines 201+)

Reviewer 2: I don't think that some overlap with option description at bottom is bad here. General description of  options and listing options relevant for specific uses case with explanation why we recommend changing them are two different things.

Author: > General description of options and listing options relevant for specific uses case with explanation why we recommend changing them are two different things

Two different things that  can be combined. For example

```
**<Config name>**
[Laconic description]
[Recommendations and personal notes]

```

Reviewer 2: I still disagree. It doesn't help anyone. For people who are not packagers, it add extra text they are not interested in, making them even less likely to read anything. Also some the recommendations might be completely opposite of their use-case.  As for people doing packaging it splits the information into multiple places making it easier to miss the important parts. Single  motivation might apply to group of options. That allows doing:
```
Use case X:
* Due to `Y` keep `A` `B` `C` on.
* Turn off `E` `F` due to `Z`

General option description
* A - does a
* B - does b
* C - does c
```

Instead of producing different kind of duplication
```
* Option  A - does a. In use case `X` due to `Y` turn on.
* Option  B - does b. In use case `X` due to `Y` turn on.
* Option  C - does c. In use case `X` due to `Y` turn on.
```
